README.md
# OpenRoulette Bot
Telegram bot with roulette, dice and orlanka game. Powered by random.org
### [@open_roulette_bot. Try it :)](https://t.me/open_roulette_bot)


![img](assets/img.png)


changelog.txt
Ченджлог (опис коммітів):
2.5.0; 17.07.2022 - Створення ченджлогу, відображення версії бота у команді /ping, створення команди /changelog, оновлення log()
2.4.1; 16.07.2022 - Видалення функції .format() в строк, де вона не використовується
2.4.0; 02.07.2022 - Оновлення великого шматку коду
2.3.0; 23.04.2022 - Додана команда /rnd
2.2.1; 22.04.2022 - Я не знаю що тут написати. Це не має сенсу
2.2.0; 22.04.2022 - Оновив readme.md
2.2.0; 22.04.2022 - Логи тепер зберігаються у папку
2.1.1; 21.04.2022 - Додан параметр minimum та maximum для функції randomorg_parse
2.1.0; 20.04.2022 - Замінено бібліотеку requests на aiohttp (надзвичане покращення швидкості)
2.0.5; 19.04.2022 - 2. Перенесення великих строк у config.py
2.0.4; 19.04.2022 - Додано .gitignore
2.0.4; 19.04.2022 - Додано reqirements.txt
2.0.4; 19.04.2022 - 1. Перенесення великих строк у config.py + Додано опцію enablelog
2.0.3; 16.04.2022 - Оптимізація за рахунок додавання функції logheader + додавання функції keyboardheader
2.0.2; 13.04.2022 - Покращення читабельності та логічності коду (2 комміти)
2.0.1; 12.04.2022 - Оптимізація імпортів
2.0.0; 11.04.2022 - Переклад Українською (!)
1.4.2; 31.12.2021 - Загальні покращення. Нічого важливого
1.4.1; 30.12.2021 - Додавання команди /help, видалення /dev та іншого лайна
1.4.0; 30.12.2021 - Покращення коди клавіатури та додавання команди /rm_keyboard
1.3.9; 30.12.2021 - Відправка "Я вас не зрозумів"
1.3.8; 30.12.2021 - Видалення великого шматка лайна з main.py
1.3.7; 26.10.2021 - Красиве відображення лого у файлі
1.3.6; 20.08.2021 - Додано команду /dev
1.3.5; 09.08.2021 - Щось незрозміле
1.3.4; 08.08.2021 - Нормальний текст логу
1.3.3; 08.08.2021 - Доданий логотип при старті боту
1.3.2; 04.08.2021 - Заміна кнопки дісклеймера на випадковий факт
1.3.1; 04.08.2021 - Додана команда /fact
1.3.0; 04.08.2021 - Бота перероблено для роботи з random.org (!) + оновлення readme.md (2 комміти)
1.2.2; 02.08.2021 - Додавання номерів ігор (для логу) та інші зміни
1.2.1; 01.08.2021 - Зрозумілий лог гри в орлянку
1.2.0; 01.08.2021 - Додано /dice та /orlanka (!)
1.1.3; 30.05.2021 - Оновлення команди /roll
1.1.2; 30.05.2021 - Додавання фото профілю боту у readme.md
1.1.2; 30.05.2021 - Cтворення readme.md
1.1.2; 30.05.2021 - Виправлення зберігання емодзі у логах та виношення токену бота у config.py
1.1.1; 29.05.2021 - Систему логування перероблено на лог у файл
1.1.0; 29.05.2021 - Систему логування створено (лог у групу (сповіщення телеграму))
1.0.1; 28.05.2021 - Виправлення помлки у main.py
1.0.0; 28.05.2021 - Бота завантаженно на github.com
1.0.0; 28.05.2021 - Бота створено


config.py
TOKEN = "bot_token"
# options
enablelog = True
howmanyfacts = 5
logfoldername = "logs"  # "." if you don't want any folder
logfilename_date = "%d.%m.%y_%H.%M"  # 21.04.22_18.17


# long strings
helpmsg = "<b>Ось список доступних команд:</b>\n\n" \
          "<b>/roll</b> - Грати в рулетку (Також є на клавіатурі)\n\n" \
          "<b>/orlanka</b> - Грати в орлянку (орел або решка)\n\n" \
          "<b>/dice</b> - Підкинути кістки\n\n" \
          "<b>/fact</b> - Випадковий факт про цього робота (Також є на клавіатурі)\n\n" \
          "<b>/disclaimer</b> - Відмова від відповідальності\n\n" \
          "<b>/author</b> - Автор бота\n\n" \
          "<b>/ping</b> - Понг!\n\n" \
          "<b>/keyboard</b> - Відкрити заново клавіатуру для гри\n\n" \
          "<b>/rm_keyboard</b> - Закрити клавіатуру. Корисно у групах (Також є на клавіатурі)\n\n" \
          "<b>/help</b> - Показати всі команди"

disclaimer_text = "<b>ДИСКЛЕЙМЕР (ВІДМОВА ВІД ВІДПОВІДАЛЬНОСТІ):" \
                " </b>Я повністю відмовляюся від результатів використання даного бота не з метою розваги." \
                " Бот створений не з комерційною метою, і ніколи таким не " \
                "стане!\n\n<b>Коротко:</b> Ставок нема."


fact1 = "Цей бот є FOSS проектом. Це означає, що його" \
        " <a href='github.com/KUKURUZKA165/roulette-telegram-bot'>вихідний код</a>" \
        " відкритий усім охочим. Будь-хто може перевірити чесність його роботи :)"

fact2 = "Випадковість у цьому роботі реалізована через атмосферний шум (random.org)." \
        " Сучасні технології не здатні передбачити результат вашої гри."

fact3 = f"Бот знає тільки {howmanyfacts} фактів себе." \
        f" Він просто перебирає факти випадково і відправляє вам."

fact4 = "Цей факт ще не придумали. Я не знаю що тут написати("

fact5 = "Навіть якби всі казино використовували такий самий спосіб отримання" \
        " випадкових чисел, як у нас, вони все одно залишалися б у плюсі."

rnd_no_args = "<b>Ви не вказали жодних параметрів для цієї команди.</b>\n\nСтруктура параметрів має бути такою:" \
              "\n<i><b>/rnd мін макс</b></i>, де:\n<i><b>мін</b></i> - Мінімальне число" \
              "\n<i><b>макс</b></i> - Максимальне число\n\n<b>Приклад:</b>\n<i><b>/rnd 1 100</b></i>"

rnd_something_went_wrong = "<b>Щось пішло не так :(</b>\nБудь ласка переконайтеся, що:\n\n<b>1. </b>Мінімальне та " \
                           "максимальне число має бути менше мільярда.\n<b>2.</b> Максимальне значення" \
                           " має бути більше мінімального значення."


algorithm.py
def fun_result(rnd):
    result = "Помилка. Скоріше за все проблема на стороні random.org"
    if rnd == 1:
        result = "Зеро! 🟢 Зелений"
    if rnd == 2:
        result = "32, 🔴 Червоний"
    if rnd == 3:
        result = "15, ⚫ Чорний"
    if rnd == 4:
        result = "19, 🔴 Червоний"
    if rnd == 5:
        result = "4, ⚫ Чорний"
    if rnd == 6:
        result = "21, 🔴 Червоний"
    if rnd == 7:
        result = "2, ⚫ Чорний"
    if rnd == 8:
        result = "25, 🔴 Червоний"
    if rnd == 9:
        result = "17, ⚫ Чорний"
    if rnd == 10:
        result = "34, 🔴 Червоний"
    if rnd == 11:
        result = "6, ⚫ Чорний"
    if rnd == 12:
        result = "27, 🔴 Червоний"
    if rnd == 13:
        result = "13, ⚫ Чорний"
    if rnd == 14:
        result = "36, 🔴 Червоний"
    if rnd == 15:
        result = "11, ⚫ Чорний"
    if rnd == 16:
        result = "30, 🔴 Червоний"
    if rnd == 17:
        result = "8, ⚫ Чорний"
    if rnd == 18:
        result = "23, 🔴 Червоний"
    if rnd == 19:
        result = "10, ⚫ Чорний"
    if rnd == 20:
        result = "5, 🔴 Червоний"
    if rnd == 21:
        result = "24, ⚫ Чорний"
    if rnd == 22:
        result = "16, 🔴 Червоний"
    if rnd == 23:
        result = "33, ⚫ Чорний"
    if rnd == 24:
        result = "1, 🔴 Червоний"
    if rnd == 25:
        result = "20, ⚫ Чорний"
    if rnd == 26:
        result = "14, 🔴 Червоний"
    if rnd == 27:
        result = "31, ⚫ Чорний"
    if rnd == 28:
        result = "9, 🔴 Червоний"
    if rnd == 29:
        result = "22, ⚫ Чорний"
    if rnd == 30:
        result = "18, 🔴 Червоний"
    if rnd == 31:
        result = "29, ⚫ Чорний"
    if rnd == 32:
        result = "7, 🔴 Червоний"
    if rnd == 33:
        result = "28, ⚫ Чорний"
    if rnd == 34:
        result = "12, 🔴 Червоний"
    if rnd == 35:
        result = "35, ⚫ Чорний"
    if rnd == 36:
        result = "3, 🔴 Червоний"
    if rnd == 37:
        result = "26, ⚫ Чорний"

    return result


longfunctions.py
def roll_function(rnd):
    result = "Помилка. Скоріше за все проблема на стороні random.org"
    if rnd == 1:
        result = "Зеро! 🟢 Зелений"
    if rnd == 2:
        result = "32, 🔴 Червоний"
    if rnd == 3:
        result = "15, ⚫ Чорний"
    if rnd == 4:
        result = "19, 🔴 Червоний"
    if rnd == 5:
        result = "4, ⚫ Чорний"
    if rnd == 6:
        result = "21, 🔴 Червоний"
    if rnd == 7:
        result = "2, ⚫ Чорний"
    if rnd == 8:
        result = "25, 🔴 Червоний"
    if rnd == 9:
        result = "17, ⚫ Чорний"
    if rnd == 10:
        result = "34, 🔴 Червоний"
    if rnd == 11:
        result = "6, ⚫ Чорний"
    if rnd == 12:
        result = "27, 🔴 Червоний"
    if rnd == 13:
        result = "13, ⚫ Чорний"
    if rnd == 14:
        result = "36, 🔴 Червоний"
    if rnd == 15:
        result = "11, ⚫ Чорний"
    if rnd == 16:
        result = "30, 🔴 Червоний"
    if rnd == 17:
        result = "8, ⚫ Чорний"
    if rnd == 18:
        result = "23, 🔴 Червоний"
    if rnd == 19:
        result = "10, ⚫ Чорний"
    if rnd == 20:
        result = "5, 🔴 Червоний"
    if rnd == 21:
        result = "24, ⚫ Чорний"
    if rnd == 22:
        result = "16, 🔴 Червоний"
    if rnd == 23:
        result = "33, ⚫ Чорний"
    if rnd == 24:
        result = "1, 🔴 Червоний"
    if rnd == 25:
        result = "20, ⚫ Чорний"
    if rnd == 26:
        result = "14, 🔴 Червоний"
    if rnd == 27:
        result = "31, ⚫ Чорний"
    if rnd == 28:
        result = "9, 🔴 Червоний"
    if rnd == 29:
        result = "22, ⚫ Чорний"
    if rnd == 30:
        result = "18, 🔴 Червоний"
    if rnd == 31:
        result = "29, ⚫ Чорний"
    if rnd == 32:
        result = "7, 🔴 Червоний"
    if rnd == 33:
        result = "28, ⚫ Чорний"
    if rnd == 34:
        result = "12, 🔴 Червоний"
    if rnd == 35:
        result = "35, ⚫ Чорний"
    if rnd == 36:
        result = "3, 🔴 Червоний"
    if rnd == 37:
        result = "26, ⚫ Чорний"

    return result


main.py
from time import ctime, strftime
from random import randint
from asyncio import sleep
from aiogram import Bot, Dispatcher, executor, types
from aiohttp import ClientSession
from config import *

print(r"   ____                   ____              __     __  __            ____        __ ")
print(r"  / __ \____  ___  ____  / __ \____  __  __/ /__  / /_/ /____       / __ )____  / /_")
print(r" / / / / __ \/ _ \/ __ \/ /_/ / __ \/ / / / / _ \/ __/ __/ _ \     / __  / __ \/ __/")
print(r"/ /_/ / /_/ /  __/ / / / _, _/ /_/ / /_/ / /  __/ /_/ /_/  __/    / /_/ / /_/ / /_  ")
print(r"\____/ .___/\___/_/ /_/_/ |_|\____/\____/_/\___/\__/\__/\___/    /_____/\____/\__/  ")
print(r"    /_/                                                                             ")


version = "2.5.0 (17.07.2022)"
bot = Bot(token=TOKEN)
dp = Dispatcher(bot)
logfile = f"{logfoldername}/log_{strftime(logfilename_date)}.txt"


markup1 = types.ReplyKeyboardMarkup(resize_keyboard=True, row_width=3)
markup1.add(types.KeyboardButton("Випадковий факт ❓"),
            types.KeyboardButton("Дайс 🎲"),
            types.KeyboardButton("Закрити ❌"),
            types.KeyboardButton("Грати"))


async def log(text):
    print(text)
    if enablelog is True:
        with open(logfile, "a", encoding="utf-8") as file:
            file.writelines(f"\n{text}")
            file.close()


async def logheader(msg):
    await log(f"--------------------\n{ctime()}\n"
              f"{msg.from_user.first_name} {msg.from_user.last_name} @{msg.from_user.username} id={msg.from_user.id}\n"
              f"{msg.chat.title} {msg.chat.invite_link} id={msg.chat.id}\n-")


async def randomorg_parse(minimum, maximum):
    async with ClientSession() as session:
        async with session.get(f"https://www.random.org/integers/"
                               f"?num=1&min={minimum}&max={maximum}&col=1&base=10&format=plain&rnd=new") as response:
            return int(await response.text())


@dp.message_handler(commands=["start"])
async def start(message):
    await logheader(message)
    await log("Запитаний старт")

    send_mess = f"<b>Привіт, {message.from_user.first_name}! У цьому боті ти зможеш крутити рулетку.</b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html")

    send_mess = "<b>Оскільки бот безкоштовний, він не хоститься на віддаленому сервері." \
                " Через це я не можу тримати його увімкненим 24/7. Приношу свої вибачення!</b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html")

    send_mess = "<b>Важливо для власників груп: </b>\n" \
                "В бота налаштований обмежений доступ до повідомлень у групах на рівні Telegram API." \
                " Він не зможе за вами підглядати :)\n" \
                "<b><a href='https://core.telegram.org/bots#privacy-mode'>Докладніше</a></b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html", disable_web_page_preview=True)

    send_mess = helpmsg + "\n\n\n<b>Або користуйся клавіатурою з кнопками:</b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html")

    send_mess = "<b>🇺🇦 Слава Україні! 🇺🇦</b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html", reply_markup=markup1)

    await log("Старт надіслано!")


@dp.message_handler(commands=["roll"])
async def roll(message):
    await logheader(message)
    send_mess = "<b>Очікування відповіді від random.org</b>"
    roll_mess = await bot.send_message(message.chat.id, send_mess, parse_mode="html", disable_web_page_preview=True)
    gamecode = randint(100, 999)
    await log(f"{gamecode} - Звернення до random.org...")

    num = await randomorg_parse(0, 36)
    if num == 0:
        result = "Зеро! 🟢 Зелений"
    elif num == 1 or num == 3 or num == 5 or num == 7 or num == 9 or num == 12 or num == 14 or num == 16 or num == 18\
            or num == 19 or num == 21 or num == 23 or num == 25 or num == 27 or num == 30 or num == 32 or num == 34\
            or num == 36:
        result = f"{num}, 🔴 Червоний"
    elif num == 2 or num == 4 or num == 6 or num == 8 or num == 10 or num == 11 or num == 13 or num == 15 or num == 17\
            or num == 20 or num == 22 or num == 24 or num == 26 or num == 28 or num == 29 or num == 31 or num == 33\
            or num == 35:
        result = f"{num}, ⚫ Чорний"
    else:
        result = "Помилка. Скоріше за все проблема на стороні random.org"

    await log(f"{gamecode} - Результат - {result}.")
    await bot.edit_message_text(chat_id=message.chat.id, message_id=roll_mess.message_id,
                                text=f"<b>{message.from_user.first_name}, ваш результат:\n{result}</b>",
                                parse_mode="html", disable_web_page_preview=True)
    await log(f"{gamecode} - Надіслано.")


@dp.message_handler(commands=["rnd"])
async def rnd_command(message):
    await logheader(message)
    await log("Запитаний rnd.")

    if message.text == "/rnd":
        await bot.send_message(message.chat.id, rnd_no_args, parse_mode="html")
        await log("Людина написала команду без аргументів.")
    else:
        minmax = message.text.replace("/rnd ", "")
        min_num, max_mun = minmax.split()
        min_num_int, max_mun_int = int(min_num), int(max_mun)

        send_mess = "<b>Очікування відповіді від random.org</b>"
        rnd_mess = await bot.send_message(message.chat.id, send_mess, parse_mode="html")

        if min_num_int >= 1000000000 or max_mun_int >= 1000000000 or min_num > max_mun:
            await bot.edit_message_text(chat_id=message.chat.id, message_id=rnd_mess.message_id,
                                        text=rnd_something_went_wrong, parse_mode="html")
            await log("Людина неправильно вказала аргументи.")
        else:
            result = await randomorg_parse(min_num, max_mun)
            await bot.edit_message_text(chat_id=message.chat.id, message_id=rnd_mess.message_id,
                                        text=f"<b>Мінімальне число: {min_num}\nМаксимальне число: {max_mun}\n\n"
                                             f"Результат: {result}</b>", parse_mode="html")
            await log(f"Вийшло {result} из {min_num} на {max_mun}")


@dp.message_handler(commands=["ping"])
async def ping(message):
    await logheader(message)
    await log("Пінг?")
    send_mess = f"<b>Понг! Я живий!</b>\n\nМоя версія:\n<code>{version}</code>\nНадішліть /changelog щоб побачити" \
                f"повний ченджлог."
    await bot.send_message(message.chat.id, send_mess, parse_mode="html")
    await log("Понг!")


@dp.message_handler(commands=["changelog"])
async def changelog(message):
    await logheader(message)
    await log("Ченджлог?")
    changelogfile = open("changelog.txt", "r", encoding="utf-8")
    await bot.send_message(message.chat.id, changelogfile.read())
    changelogfile.close()
    await log("Ченджлог!")


@dp.message_handler(commands=["disclaimer"])
async def disclaimer(message):
    await logheader(message)
    await log("Запитаний дисклеймер.")
    await bot.send_message(message.chat.id, disclaimer_text, parse_mode="html")
    await log("Дисклеймер відправлено.")


@dp.message_handler(commands=["author"])
async def author(message):
    await logheader(message)
    await log("Запрошений автор.")
    send_mess = "<b>🧑🏻‍💻 Мій автор - @anton165\nЙому можна давати ідеї для нових функцій у боті.</b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html")
    await log("Автор засланий)")


@dp.message_handler(commands=["keyboard"])
async def keyboard(message):
    await logheader(message)
    await log("Запрошено відкриття клавіатури!")
    send_mess = "<b>Клавіатура відкрита!</b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html", reply_markup=markup1)
    await log("Клавіатура відкрита.")


@dp.message_handler(commands=['rm_keyboard'])
async def rm_keyboard(message):
    await logheader(message)
    await log("Запитано закриття клавіатури!")
    send_mess = f"<b>Клавіатура закрита!</b>\nДля відкриття напишіть <b>/keyboard</b>"
    await bot.send_message(message.chat.id, send_mess, parse_mode="html", reply_markup=types.ReplyKeyboardRemove())
    await log("Клавіатура закрита.")


@dp.message_handler(commands=["orlanka"])
async def orlanka(message):
    await logheader(message)
    gamecode = randint(10, 99)
    send_mess = "<b>Очікування відповіді від random.org</b>"
    oreshka_mess = await bot.send_message(message.chat.id, send_mess, parse_mode="html", disable_web_page_preview=True)
    await log(f"{gamecode} - Запитана орлянка.")
    oreshka = await randomorg_parse(1, 2)

    resultoreshka = "Помилка. Скоріше за все проблема на стороні random.org"
    if oreshka == 1:
        await log(f"{gamecode} - Випав орел. Відправляю...")
        resultoreshka = "Орел"
    elif oreshka == 2:
        await log(f"{gamecode} - Випала решка. Відправляю...")
        resultoreshka = "Решка"

    await bot.edit_message_text(chat_id=message.chat.id, message_id=oreshka_mess.message_id,
                                text=f"<b>{message.from_user.first_name}, ваш результат:\n{resultoreshka}.</b>",
                                parse_mode="html")
    await log(f"{gamecode} - Надіслано.")


@dp.message_handler(commands=["dice"])
async def dice(message):
    await logheader(message)
    gamecode = randint(10, 99)
    dice_sleep = 3
    await log(f"{gamecode} - Запрошений дайс.")
    dice_message = await bot.send_dice(message.chat.id, emoji="🎲")
    await log(f"{gamecode} - Підкинув. Результат: {dice_message.dice.value}. Чекаємо {dice_sleep} сек...")
    send_mess = "<b>Зачекайте, будь ласка...</b>"
    dice_comment_message = await bot.send_message(message.chat.id, send_mess, parse_mode="html")

    dice_emoji = ""
    if dice_message.dice.value == 1:
        dice_emoji = "⚀"
    elif dice_message.dice.value == 2:
        dice_emoji = "⚁"
    elif dice_message.dice.value == 3:
        dice_emoji = "⚂"
    elif dice_message.dice.value == 4:
        dice_emoji = "⚃"
    elif dice_message.dice.value == 5:
        dice_emoji = "⚄"
    elif dice_message.dice.value == 6:
        dice_emoji = "⚅"

    await sleep(dice_sleep)
    await bot.edit_message_text(chat_id=message.chat.id, message_id=dice_comment_message.message_id,
                                text=f"<b>{message.from_user.first_name}, ваш результат:\n{dice_emoji}"
                                     f" {dice_message.dice.value}</b>", parse_mode="html")
    await log(f"{gamecode} - Надіслано.")


@dp.message_handler(commands=["fact"])
async def fact(message):
    await logheader(message)
    await log("Запитаний факт.")
    send_fact_mess = "Виникла помилка!"
    rnd_fact = randint(1, howmanyfacts)

    if rnd_fact == 1:
        send_fact_mess = fact1
    elif rnd_fact == 2:
        send_fact_mess = fact2
    elif rnd_fact == 3:
        send_fact_mess = fact3
    elif rnd_fact == 4:
        send_fact_mess = fact4
    elif rnd_fact == 5:
        send_fact_mess = fact5

    await bot.send_message(message.chat.id, "<b>Цікавий факт:</b>\n" + send_fact_mess, parse_mode="html",
                           disable_web_page_preview=True)

    await log(f"Я надіслав факт {rnd_fact} з {howmanyfacts}. Його зміст:\n{send_fact_mess}")


@dp.message_handler(commands=["help"])
async def help_command(message):
    await logheader(message)
    await log("Запитана допомога")
    send_mess = helpmsg
    await bot.send_message(message.chat.id, send_mess, parse_mode="html")
    await log("Готово")


@dp.message_handler(content_types=["text"])
async def mess(message):
    get_message_bot = message.text.strip().lower()
    if get_message_bot == "грати":
        await log("--------------------\nКлавіатура:")
        await roll(message)
    elif get_message_bot == "закрити ❌":
        await log("--------------------\nКлавіатура:")
        await rm_keyboard(message)
    elif get_message_bot == "дайс 🎲":
        await log("--------------------\nКлавіатура:")
        await dice(message)
    elif get_message_bot == "випадковий факт ❓":
        await log("--------------------\nКлавіатура:")
        await fact(message)
    else:
        await log(f"--------------------\n{ctime()}\n{message.from_user.first_name} {message.from_user.last_name}"
                  f" @{message.from_user.username} id={message.from_user.id}\n{message.chat.title}"
                  f" {message.chat.invite_link} id = {message.chat.id}\n-\nНадіслав(ла): {message.text}")
        if message.from_user.id == message.chat.id:
            send_mess = "<b>Я вас не зрозумів :(</b>"
            await bot.send_message(message.chat.id, send_mess, parse_mode="html")
            await log("Написав користувачеві, що я його не зрозумів")

if __name__ == '__main__':
    print("Доброго дня, Слава Україні!")
    executor.start_polling(dp)
